glusterd2/daemon/daemon.go

```diff
@@ -52,11 +52,12 @@ type Daemon interface {
 // When wait == true, this function can be used to spawn short term processes
 // which will be waited on for completion before this function returns.
 func Start(d Daemon, wait bool) error {
+	args := strings.Fields(d.Args())
 
 	log.WithFields(log.Fields{
 		"name": d.Name(),
 		"path": d.Path(),
-		"args": d.Args(),
+		"args": args,
```

Comment:
Sorry, but could you explain how only args instead of d.Args() help in fixing this issue?

Reply:
The method Args() in brick/glusterfsd.go assigns a port and then returns it along with other arguments required to start the daemon as a string. Now calling d.Args() two times means port allocation happens twice while trying to start the daemon.

Reply:
I think you can fix `Args()` itself. Add below lines in the beginning of `Args()` function,

    if b.args != "" {
        return b.args
    }

Reply:
Making changes in glusterfsd.Args() is causing volume start to fail in the functional tests. Having these changes in daemon.Start() however doesn't fail these tests. I'm at a lack for any logical explanation behind this.